### PR TITLE
Plugins can accept object as options

### PR DESCRIPTION
**Pull request by lukefx**
Please fill in this template.

- [X] Prefer to make your PR against the `types-2.0` branch.
- [X] The package does not provide its own types, and you can not add them.
- [X] Test the change in your own code.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).

If adding a new definition:
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Run `tsc` without errors.
- [X] Include the required [files](https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package) and header.

If changing an existing definition:
- [X] Provide a URL to  documentation or source code which provides context for the suggested changes: [https://github.com/selectize/selectize.js/blob/master/docs/plugins.md#list-with-options](https://github.com/selectize/selectize.js/blob/master/docs/plugins.md#list-with-options)
- [x] Increase the version number in the header if appropriate.

### CODE REVIEW

File: selectize/index.d.ts [hunk 1]
```diff
@@ -155,7 +155,7 @@ declare namespace Selectize {
          *
          * Default: null
          */
-        plugins?: string[];
+        plugins?: string[] | Object;
```
Comment (adidahiya) [inline selectize/index.d.ts]: please be more specific and define this interface. `Object` is too broad of a type.

**Review by adidahiya — CHANGES_REQUESTED** [review]

File: selectize/index.d.ts (hunk 1 above)
Comment (lukefx) [inline selectize/index.d.ts]: It's almost impossibile, every plugins can have different key for properties...

````
$('select').selectize({
    plugins: {
        'plugin_a': { /* ... */ },
        'plugin_b': { /* ... */ }
    }
});```

Comment (vvakame) [inline selectize/index.d.ts]: use `any` instead of `Object` at least.

**Comment by paulvanbrenk** [comment]
This PR seems abandoned, please re-open when you made the requested changes.

**Pull request closed without merging**